tests/unit/deposit/test_api_simpleflow.py

```diff
@@ -48,9 +50,24 @@ def make_file_fixture(filename, text=None):
     return (BytesIO(content), filename)
 
 
-def test_simple_rest_flow(api, api_client, db, es, location, users,
+class MockDataCiteProvider(DataCiteProvider):
```

Comment:
I think its better to mock ``invenio_pidstore.providers.datacite.DataCiteMDSClient`` instead. This way we insure that the DOI is marked as registered in PIDStore.